lib/config/index.js

```diff
@@ -54,9 +46,22 @@ const exists = fs.existsSync || path.existsSync
 let logger = null // Lazy-loaded in `initialize`.
 let _configInstance = null
 
+const getConfigFileNames = () => [
+  process.env.NEW_RELIC_CONFIG_FILENAME,
```

Comment:
You could keep the file name config array and file locations arrays as constants at the top and do something like `CONFIG_FILE_LOCATIONS.filter(Boolean).splice` rather than using a function.

Reply:
as I mentioned, this was changed for testing purposes. Since this was constructed at load time I was unable to modify these in my tests without changing them to functions instead of just constant lists

Reply:
Were there other tests failing? I just tried the `config.test.js` tests with the arrays directly, rather than from a function and the tests pass. Sorry to be a pain, but could you try the tests again with `CONFIG_FILE_LOCATIONS` and a `CONFIG_FILE_NAMES` array?

Reply:
with the file names as an array instead of a function:

![image](https://user-images.githubusercontent.com/5516565/118169945-d1b6a980-b429-11eb-9fd6-7641a2d57f7b.png)

The old tests work fine, my new ones are the ones that are failing since the ENV-vars are read when the module is loaded which make them impossible to override in my tests.

Reply:
The thing is, if they are not functions, they will default to the current working directory and search for newrelic.js, that file exists which will make them pass since the lists are constructed att module load time.

I changed the assertions so we have the supposed filename within the config so we can assert against that. 

All in all, they need to be functions since we need to change the values of those lists in the tests otherwise we would just search for newrelic.js in the root of the project. Now they will fail the way they are supposed to since I added something better to assert against within the config.